### PR TITLE
Added sysctl utility as with key value , key  can be

### PR DESCRIPTION
**Pull request by PraveenPenguin**
set with new vlaue

1- Generic implementation of sysctl, to read and write
2- implementation of sysctl, for kernel params

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
@clebergnu can you please have look on this PR

File: avocado/utils/linux_utils.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+"""
+Linux OS  utilities
+"""
+
+import re
+
+from . import genio
+
+
+def sysctl(key, value=None):
```
Comment (clebergnu) [inline avocado/utils/linux_utils.py]: I find it highly uncommon to have a single function for both setting and getting a value... I honestly think I can see things from your perspective, in the sense that this is such a simple function that it's not worth having two.

But, one of the Python motto (in the Zen of Python) is that explicit is better than implicit.  So, I'd suggest you to define both a `get_proc_sys()` and `set_proc_sys()` types of function.

File: avocado/utils/linux_utils.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    :return: The single-line sysctl value as a string.
+    """
+    path = '/proc/sys/%s' % key
+    if value is not None:
+        genio.write_one_line(path, str(value))
+    return genio.read_one_line(path)
+
+
+def sysctl_kernel(key, value=None):
```
Comment (clebergnu) [inline avocado/utils/linux_utils.py]: And then here we have an opportunity to not have this function, but instead, just add `kernel` to the key, that if you think "compound" keys are acceptable.

File: avocado/utils/linux_utils.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+def sysctl(key, value=None):
+    """Generic implementation of sysctl, to read and write.
+
+    :param key: A location under /proc/sys
+    :param value: If not None, a value to write into the sysctl.
+
+    :return: The single-line sysctl value as a string.
+    """
+    path = '/proc/sys/%s' % key
```
Comment (clebergnu) [inline avocado/utils/linux_utils.py]: `os.path.join()` is preferable.

File: avocado/utils/linux_utils.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+def sysctl_kernel(key, value=None):
+    """implementation of sysctl, for kernel params"""
+    if value is not None:
+        # write
+        genio.write_one_line('/proc/sys/kernel/%s' % key, str(value))
+    else:
+        # read
+        out = genio.read_one_line('/proc/sys/kernel/%s' % key)
+        return int(re.search(r'\d+', out).group(0))
```
Comment (clebergnu) [inline avocado/utils/linux_utils.py]: Why are you defining the output to be integer only?  For instance,

```
$ cat /proc/sys/kernel/version
#1 SMP Mon Apr 8 15:40:59 UTC 2019
```

Would not be supported.

File: avocado/utils/linux_utils.py (hunk 1 above)
Comment (PraveenPenguin) [reply avocado/utils/linux_utils.py]: @clebergnu yes make sense will adopt this

File: avocado/utils/linux_utils.py (hunk 2 above)
Comment (PraveenPenguin) [reply avocado/utils/linux_utils.py]: yes good point , let me try that

File: avocado/utils/linux_utils.py (hunk 4 above)
Comment (PraveenPenguin) [reply avocado/utils/linux_utils.py]: ah i did not notice this , will check

**Comment by PraveenPenguin** [comment]
@clebergnu thanks for review will adopt the changes

File: avocado/utils/linux_utils.py (hunk 4 above)
Comment (clebergnu) [reply avocado/utils/linux_utils.py]: Hi @PraveenPenguin, I put the comments together on a v2: https://github.com/avocado-framework/avocado/pull/3138

Your review is very much appreciated.  Thanks

**Pull request closed without merging**